setup.py

```diff
@@ -110,48 +138,72 @@ def run_tests(self):
+    try:
+        init = fp.read()
+        __version__ = re.search(r'__version__\s*=\s*[\'"]([^\'"]*)[\'"]', init).group(1)
+        __author__ = re.search(r'__author__\s*=\s*[\'"]([^\'"]*)[\'"]', init).group(1)
+        __author_email__ = re.search(r'__author_email__\s*=\s*[\'"]([^\'"]*)[\'"]', init).group(1)
+        __license__ = re.search(r'__license__\s*=\s*[\'"]([^\'"]*)[\'"]', init).group(1)
+        __status__ = re.search(r'__status__\s*=\s*[\'"]([^\'"]*)[\'"]', init).group(1)
+        __url__ = re.search(r'__url__\s*=\s*[\'"]([^\'"]*)[\'"]', init).group(1)
+    except IndexError:
+        raise RuntimeError('Unable to determine details of package.')
 
+with open_local(['README.rst']) as fr:
+    long_description = fr.read()
 
 if sys.version_info.major != 3:
     printer(INSTALLATION_ERROR.format(sys.version_info))
     sys.exit(1)
 
-VERSION = _get_version()
-
 with open('requirements.txt') as f:
     REQUIRED = f.read().splitlines()
 
-setup(
-    name='BiblioPixel',
-    version=VERSION,
-    description=(
+setup_kwargs = {
+    'name': 'BiblioPixel',
+    'version': __version__,
+    'description': [
         'BiblioPixel is a pure python library for manipulating a wide variety '
-        'of LED strip based displays, both in strip and matrix form.'),
-    long_description=open('README.rst').read(),
-    author='Adam Haile',
-    author_email='adam@maniacallabs.com',
-    url='http://github.com/maniacallabs/bibliopixel/',
-    license='MIT',
-    packages=find_packages(exclude=['test']) + ['ui', 'scripts'],
-    classifiers=[
+        'of LED strip based displays, both in strip and matrix form.'
+    ],
+    'long_description': long_description,
+    'author': __author__,
+    'author_email': __author_email__,
+    'url': __url__,
+    'license': __license__,
+    'packages': find_packages(exclude=['test']) + ['ui', 'scripts'],
+    'classifiers': [
         'Development Status :: 5 - Production/Stable',
         'License :: OSI Approved :: MIT License',
         'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.6'
     ],
-    tests_require=['pytest'],
-    cmdclass={
+    'tests_require': ['pytest'],
+    'cmdclass': {
         'benchmark': RunBenchmark,
+        'clean': SetupClean,
         'coverage': RunCoverage,
         'test': RunTests,
-        'install_scripts': InstallScripts
+        'install_scripts': InstallScripts,
+        'install': SetupInstall
     },
-    include_package_data=True,
-    scripts=['scripts/bp', 'scripts/bibliopixel'],
-    install_requires=REQUIRED
-)
+    'include_package_data': True,
+    # 'scripts': ['scripts/bp', 'scripts/bibliopixel'],
+    'entry_points': {
+        'console_scripts': ['bibliopixel = bibliopixel.__main__:main', 'bp = bibliopixel.__main__:main']
+    },
+    'install_requires': REQUIRED
+}
+
+setup(**setup_kwargs)
```

Comment:
What's the advantage to creating this new `setup_kwargs` variable?

Reply:
Personal preference, and can be reverted.